modules/ROOT/pages/dataweave-formats.adoc

```diff
@@ -80,42 +80,49 @@ DataWeave supports the following formats as input and output:
 [[dw_readers_writers]]
 == DataWeave Readers
 
-DataWeave can read input data as a whole in-memory, in indexed fashion,
-and for some data formats, part-by-part by streaming the input. When attempting
-to read a large file, it is possible to run out of memory or to impact
-performance negatively. Streaming can improve performance but impacts access to file.
-
-* Indexed and In-Memory: +
-Allow for random access to data because both
-strategies parse the entire document. When using these strategies, a DataWeave
-script can access any part of the resulting value at any time.
-
-** Indexed: +
-Uses indexes over the disk.
-+
-DataWeave can only read the following formats in the indexed mode:
-+
-CSV, JSON, XML (starting in Mule 4.3.0)
-+
-Note that DataWeave _does not_ support reading Excel (XLSX) input in the indexed mode.
-
-** In-Memory: +
-Parses the entire document in memory.
-+
-DataWeave can read all supported formats using the in-memory mode.
+DataWeave can read input data as a whole by loading it into memory or by indexing it in local storage and, for some data formats, read data sequentially in parts by streaming the input. The indexed reading strategy implements a mechanism to avoid out-of-memory issues when processing large files. The streaming strategy can improve performance but impacts access to file.
 
 [[streaming]]
-* Streaming: +
-Allows for sequential access to the file. This strategy partitions
+[%header%autowidth.spread,cols="a,a,a", options="header"]
+|===
+| Read Strategy | Description | Supported Formats
+<.^| In-Memory
+| Enables random access to data by parsing the entire document. When using this strategy, a DataWeave script can access any part of the resulting value at any time.
+| DataWeave can read all supported formats using this strategy.
+<.^| Indexed
+| Enables random access to data by parsing the entire document. When using this strategy, a DataWeave script can access any part of the resulting value at any time.
+
+When processing a `String` with a size larger than 1.5 MB, DataWeave automatically splits the value in chunks to avoid out-of-memory issues. This feature works only with `JSON` and `XML` input data.
+
+You can configure the maximum size DataWeave uses to split strings by modifying the following system property:
+
+```
+com.mulesoft.dw.max_memory_allocation
+```
+//can we show an example of a valid value? Is 4096 (chars) the default?
+
+Note that using this feature has a negative impact on performance due to splitting strings and accessing them through local storage. If your environment has enough resources to load all content in memory, you can also disable this string management feature completely by setting the value of the following system property to `false`:
+
+```
+com.mulesoft.dw.buffered_char_sequence.enabled
+```
```

Comment:
We were discussing if this behavior should be documented here and maybe we could add this to the dataweave-memory-management section. Because these details seem to fit more there than in this general page.